### ISSUE TITLE
Regex + transformations are not supported?

### ISSUE THREAD

**Issue by ahmedelgabri** [hrsh7th/vim-vsnip#83]
I can't get this to work it throws an error, `$TM_FILENAME_BASE` by itself works fine

```json
{
  "test": {
    "prefix": "test",
    "body": "${TM_FILENAME_BASE/(.*)/${1:/capitalize}/}"
  }
}
```

```text
Error detected while processing function
<SNR>101_expand[1]
vsnip#expand[7]
vsnip#anonymous[4]
33[1]
17[1]
vsnip#snippet#parser#parse[7]
E605: Exception not caught: {"result": [false, null, 0], "text": "${TM_FILENAME_BASE/(.*)/${1:/capitalize}/}"}
Error detected while processing function
<SNR>101_expand[1]
vsnip#expand[7]
E171: Missing :endif
```

**Comment by hrsh7th** [comment]
Sorry.
Currently, vsnip does not supported the transform feature.

transform feature is hard to implement because we should parse and translate javascript regexp.

**Comment by hrsh7th** [comment]
Anyway, The parse failing does not expect so I've fixed it.

**Comment by hrsh7th** [comment]
Hm... we can't support `[^\W]` regex maybe...

**Comment by ahmedelgabri** [comment]
> Anyway, The parse failing does not expect so I've fixed it.

Fixed, thanks

> transform feature is hard to implement because we should parse and translate javascript regexp.

I understand that, but if the project goal is to support VSCode snippets format, IMO this is required. Otherwise, you can support vim regex instead but this will be breaking the interoperability with VSCode snippets then.

**Comment by hrsh7th** [comment]
Yes. I planning and investigating to support it.

The feature is hard to support but I think the implementation is fun.

**Comment by jandamm** [comment]
Since ${VIM:...} is supported, is there a way to use Regex this way? At least using Vims Regex (maybe `\v` by default).

**Comment by jimcornmell** [comment]
As a workaround is it possible to pass variables into VIM/python, and then use python to manipulate them?

```
${VIM:system('python -c \"import sys,re;sys.stdout.write(re.sub(r'[^0-9^a-z]', "",$TM_FILENAME))\"')}
```

I know its not portable, but neither is $VIM....

Then we can use this: https://www.tutorialspoint.com/python/python_reg_expressions.htm

**Comment by nawordar** [comment]
Couldn't `/upcase`/`/downcase`/`/capitalize` be implemented first? It seems to be much easier to implement than Regex transformation support.

**Comment by thorlucas** [comment]
I agree that this is a very important thing to support. I believe UltiSnips manages to do it but they have the luxury of working with Python.
